ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/sync/tasks/CompleteBlocksTaskTest.java

```diff
@@ -79,6 +95,115 @@ public void shouldCompleteWithoutPeersWhenAllBlocksAreEmpty() {
+    assertThat(task.run()).isCompletedWithValue(expectedBlocks);
+  }
+
+  @Test
+  public void shouldCompleteWhenBlocksAreEmptyExceptForWithdrawals() {
+    final ProtocolSchedule mockProtocolSchedule = Mockito.mock(ProtocolSchedule.class);
+    final ProtocolSpec mockParisSpec = Mockito.mock(ProtocolSpec.class);
+    final ProtocolSpec mockShanghaiSpec = Mockito.mock(ProtocolSpec.class);
+    final WithdrawalsProcessor mockWithdrawalsProcessor = Mockito.mock(WithdrawalsProcessor.class);
+
+    final Withdrawal withdrawal =
+        new Withdrawal(UInt64.ONE, UInt64.ONE, Address.fromHexString("0x1"), GWei.ONE);
+    final List<Withdrawal> withdrawals = List.of(withdrawal);
+    final Hash withdrawalsRoot = BodyValidation.withdrawalsRoot(withdrawals);
+
+    final BlockHeader preWithdrawalsHeader = new BlockHeaderTestFixture().number(1).buildHeader();
+    final BlockHeader firstWithdrawalsHeader =
+        new BlockHeaderTestFixture().number(2).withdrawalsRoot(withdrawalsRoot).buildHeader();
+    final BlockHeader secondWithdrawalsHeader =
+        new BlockHeaderTestFixture().number(3).withdrawalsRoot(Hash.EMPTY_TRIE_HASH).buildHeader();
+
+    final Block block1 = new Block(preWithdrawalsHeader, BlockBody.empty());
+    final Block block2 =
+        new Block(
+            firstWithdrawalsHeader,
+            new BlockBody(
+                Collections.emptyList(), Collections.emptyList(), Optional.of(withdrawals)));
+    final Block block3 =
+        new Block(
+            secondWithdrawalsHeader,
+            new BlockBody(
+                Collections.emptyList(),
+                Collections.emptyList(),
+                Optional.of(Collections.emptyList())));
+    final List<Block> expected = asList(block1, block2, block3);
+
+    final RespondingEthPeer respondingPeer =
+        EthProtocolManagerTestUtil.createPeer(ethProtocolManager, 1000);
+    final RespondingEthPeer.Responder responder = responderForFakeBlocks(expected);
+
+    when(mockProtocolSchedule.getByBlockHeader((eq(preWithdrawalsHeader))))
+        .thenReturn(mockParisSpec);
+    when(mockParisSpec.getWithdrawalsProcessor()).thenReturn(Optional.empty());
+    when(mockProtocolSchedule.getByBlockHeader((eq(secondWithdrawalsHeader))))
+        .thenReturn(mockShanghaiSpec);
+    when(mockShanghaiSpec.getWithdrawalsProcessor())
+        .thenReturn(Optional.of(mockWithdrawalsProcessor));
+
+    final EthTask<List<Block>> task =
+        CompleteBlocksTask.forHeaders(
+            mockProtocolSchedule,
+            ethContext,
+            List.of(preWithdrawalsHeader, firstWithdrawalsHeader, secondWithdrawalsHeader),
+            maxRetries,
+            new NoOpMetricsSystem());
+
+    final CompletableFuture<List<Block>> runningTask = task.run();
+
+    assertThat(runningTask).isNotDone();
+    respondingPeer.respond(responder);
+    assertThat(runningTask).isCompletedWithValue(expected);
+    //    assertThat(task.run()).isCompletedWithValue(expected);
```

Comment:
Left over from testing?

Reply:
hmmm, wonder if @gfukushima didn't push the latest commit, I thought we removed that

Reply:
Good catch, thanks